### ISSUE TITLE
rpoll example doesn't work

### ISSUE THREAD

**Issue by Botster** [luaposix/luaposix#375]
Machine:
openSUSE Leap 15.6
Linux 6.4.0-150600.23.7-default #1 SMP PREEMPT_DYNAMIC Fri Jun 14 14:33:11 UTC 2024 (33f31da) x86_64 x86_64 x86_64 GNU/Linux

Ref: http://luaposix.github.io/luaposix/modules/posix.poll.html

I am trying the example code given under rpoll Usage:

```
fh = io.open "one"
while true do
  r = rpoll (fh, 500)
  if r == 0 then
    print 'timeout'
  elseif r == 1 then
    print (fh:read ())
  else
    print "finish!"
    break
  end
end

```

I got:
lua: rpoll.lua:3: attempt to call global 'rpoll' (a nil value)
stack traceback:
        rpoll.lua:3: in main chunk
        [C]: ?

So, figuring it was a simple oversight, I added:
require "posix.poll"

and tried again. I got the same error (different line number):
lua: rpoll.lua:4: attempt to call global 'rpoll' (a nil value)
stack traceback:
        rpoll.lua:4: in main chunk
        [C]: ?

So then I tried with the following require:
local p = require "posix.poll"

and called rpoll() as p.rpoll()

I got:
lua: rpoll.lua:4: bad argument #1 to 'rpoll' (int expected, got userdata)
stack traceback:
        [C]: in function 'rpoll'
        rpoll.lua:4: in main chunk
        [C]: ?

What's wrong here?

**Comment by gvvaughan** [comment]
Good catch, thank you.  I guess no one has tried that example code in 12 years :-o

According to NEWS.md (found by `git grep rpoll` in my local clone of luaposix repository):

```
- It makes one small change: rpoll now uses file descriptors, not Lua file
    objects (hence, via fileno, it can use both).
```

That is, we want rpoll to function correctly with the raw file descriptors returned by other luaposix functions such as `posix.fcntl.open`, but you can convert a Lua file handle (as returned by `io.open`) to a file descriptor using `posix.stdio.fileno`.

All of the examples in the documentation assume you have imported the functions being shown from the module being documented, i.e.

```lua
local posix = require 'posix'
local rpoll = posix.poll.rpoll
```

But the example should have been updated to use `r = rpoll (fileno(fh), 500)` (where `local fileno = posix.stdio.fileno`).  Would you like to propose a PR to correct the documentation?

**Comment by Botster** [comment]
Would I like to clone the repo just to submit a PR to correct the documentation? Well, no, not really. Especially since luaposix won't even load now due to missing module bit32. I have no clue why this issue didn't surface before.

Anyway, just in case someone wants an updated example in the next 12 years, I'm pasting it below. Whether or not it works, I don't know.

```
local posix = require 'posix'
local rpoll = posix.poll.rpoll
local fileno = posix.stdio.fileno

-- mkfifo /tmp/FIFO1
fh = io.open "/tmp/FIFO1"
while true do
  r = rpoll (fileno(fh), 500)
  if r == 0 then
    print 'timeout'
  elseif r == 1 then
    print (fh:read ())
  else
    print "finish!"
    break
  end
end
```

**Comment by gvvaughan** [comment]
From the luaposix README.md
> Bugs reports and code contributions
>
>These libraries are maintained by their users.

Thank you for making the time to report this issue, although I'm very sorry to learn that you want to download and use luaposix, but that you don't want to spend 10mins making a PR to contribute back.

I spent more time explaining to you exactly how to fix the problem you are having, and showed you how I investigated and root caused the change that caused it that it would have taken me to make the fix myself.  I had hoped that you might have appreciated that enough to make a small contribution of your time and effort in return.  Know that you know how to use `rpoll` correctly that's disappointing.

Would you also like me to make time to diagnose and fix the other problem you reported now?

**Comment by Botster** [comment]
> Thank you for making the time to report this issue, although I'm very sorry to learn that you want to download and use luaposix, but that you don't want to spend 10mins making a PR to contribute back.

I actually don't want to download and use luaposix. I was only investigating it as a possible solution. I found the issue and reported it so that others wouldn't come along and get frustrated by an example that didn't work.

Even though I had decided not to use Lua (with luaposix), I was still willing to take the time to test the updated example to ensure it worked.  But, I cannot do that now as it does not work for me.

Thank you for your time.
